misoc/software/libbase/system.c

```diff
@@ -62,6 +71,13 @@ void flush_cpu_dcache(void)
 
 	for (i = 0; i < cache_size; i += cache_block_size)
 		mtspr(SPR_DCBIR, i);
+#elif defined (__vexriscv__)
+    unsigned long cacheInfo = read_csr(0xCC0);
+    unsigned long cacheWaySize = cacheInfo & 0xFFFFF;
+    unsigned long cacheLineSize = (cacheInfo >> 20) & 0xFFF;
+    for(register unsigned long idx asm ("a0") = 0;idx < cacheWaySize;idx += cacheLineSize){
```

Comment:
Register-pinned variables are not supported in clang, so rewrite this to move from a compiler-provided register to `a0`.

Also, doesn't the assembler support a mnemonic for this instruction? Or is this a custom one of vexriscv?

Reply:
What i can do is implementing it as a assembly function.

This is a custom instruction.
The RISC-V spec doesn't specify cache flush/clean instruction (On their linux capable cores they implement memory coherency)

Reply:
Just  use `asm volatile (".word ..." :: "{a0}"(idx));` which should put the index in place

Reply:
asm volatile(".word(0b01110000000001010101000000001111)" :: "x10"(idx));
         ^~~
/home/spinalvm/hdl/misoc/misoc/software/libbase/system.c:79:9: error: matching constraint references invalid operand number

Reply:
Ok, then `asm volatile("mov x10, %0 \n .word(0b01110000000001010101000000001111)"::"r"(idx))`

Reply:
Great, that work-around is fine, thanks.